lib/srv/db/cloud/watchers/rds.go

```diff
@@ -268,24 +286,84 @@ func (f *rdsAuroraClustersFetcher) String() string {
 
 // rdsFilters returns filters to make sure DescribeDBInstances call returns
 // only databases with engines Teleport supports.
-func rdsFilters() []*rds.Filter {
+func rdsFilters(ctx context.Context, rdsClient rdsiface.RDSAPI, checkFilters bool, maxPages int) ([]*rds.Filter, error) {
+	supportedEngines := []string{
+		services.RDSEnginePostgres,
+		services.RDSEngineMySQL,
+		services.RDSEngineMariaDB,
+	}
+	if !checkFilters {
+		return []*rds.Filter{{
+			Name:   aws.String("engine"),
+			Values: aws.StringSlice(supportedEngines),
+		}}, nil
+	}
+
+	engines, err := recognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	return []*rds.Filter{{
-		Name: aws.String("engine"),
-		Values: aws.StringSlice([]string{
-			services.RDSEnginePostgres,
-			services.RDSEngineMySQL,
-			services.RDSEngineMariaDB}),
-	}}
+		Name:   aws.String("engine"),
+		Values: aws.StringSlice(engines),
+	}}, nil
 }
 
 // auroraFilters returns filters to make sure DescribeDBClusters call returns
 // only databases with engines Teleport supports.
-func auroraFilters() []*rds.Filter {
+func auroraFilters(ctx context.Context, rdsClient rdsiface.RDSAPI, checkFilters bool, maxPages int) ([]*rds.Filter, error) {
+	supportedEngines := []string{
+		services.RDSEngineAurora,
+		services.RDSEngineAuroraMySQL,
+		services.RDSEngineAuroraPostgres,
+	}
+	if !checkFilters {
+		return []*rds.Filter{{
+			Name:   aws.String("engine"),
+			Values: aws.StringSlice(supportedEngines),
+		}}, nil
+	}
+
+	engines, err := recognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
```

Comment:
Could you add a comment describing describing the `recognizedRDSEngines` logic.

Reply:
sure